lib/mcb/editor/base.rb

```diff
@@ -0,0 +1,31 @@
+module MCB
+  module Editor
+    class Base
+      attr_reader :provider
+
+      def initialize(provider:, requester:)
+        @provider = provider
+        @requester = requester
+
+        setup_cli
+        check_authorisation
+      end
+
+    protected
+
+      def setup_cli
+        raise NotImplementedError, "Needs to be implemented in Child Class"
+      end
+
+      def check_authorisation
```

Comment:
Only `CoursesEditor` & `ProviderEditor` actually uses this so maybe

MCB::Editor::Base
MCB::Editor::AuthorisationBase

Reply:
It's a bit odd that `GrantAccessWizard` and `RevokeAccessWizard` doesn't require Authorisation.

Reply:
@misaka @timabell this looks odd I guessing the assumption its one of us.

so may be `requester.is_admin` should be fair

I'm not fancying  `check_authorisation` just returns turns`true` as it defeats authorisation via assumption

also see https://ukgovernmentdfe.slack.com/archives/CGVNJ2Q2F/p1566471414155000

I'm also guessing `AccessRequest` flow is now neglected...

Reply:
I think we should be adding authorisation methods to the Pundit policy.